nixos/modules/system/boot/systemd/initrd.nix

```diff
@@ -550,12 +560,50 @@ in {
             exit 0
           fi
           echo 'NEW_INIT=' > /etc/switch-root.conf
+        '';
+      };
+
+      # We need to propagate /run for things like /run/booted-system
+      # and /run/current-system.
+      mounts = [
```

Comment:
I included it in the first commit, as it is part of the change that splits the activation service into two services. I feel that this mount does not really belong in either, so that's why I made it into its own independent mount unit.